README.md

```diff
@@ -1,31 +1,283 @@
+4. Request Kody to generate API Key assigned to the Store. API key is needed for the authorization of the payment request.
+
+Once the above pre-requisite has met, you can start developing the code to make a payment request using the API key.
+
+Following steps are needed:
+
+1. **Kody SDK Client**: The Kody SDK Client allows direct communication with the Kody Payments Gateway. It simplifies the integration of Kody payments into your application.
 The SDK is generated from Protobuf gRPC protocols to facilitate communication with the Kody Payments Gateway.
+Kody SDK Client are available in following languages:
+- Java : https://github.com/KodyPay/kody-clientsdk-java/
+- Python: https://github.com/KodyPay/kody-clientsdk-python/
+- PHP: https://github.com/KodyPay/kody-clientsdk-php/
+- .Net: https://github.com/KodyPay/kody-clientsdk-dotnet/
+
+2. **Get list of Terminals** : Terminals request is made to verify the list of available terminals and to check whether it is online or not.
+
+Key attributes to make a terminal call:
+- TerminalsRequest : Request message for fetching terminal status 
+```protobuf
+message TerminalsRequest {
+  string store_id = 1; // UUID of store
+}
+```
+- KodyPayTerminalService : Service used to make the Terminal request.
+- KodyPayTerminalService.Terminals: RPC function used to make the Terminal Request
+```protobuf
+service KodyPayTerminalService {
+  rpc Terminals(TerminalsRequest) returns (TerminalsResponse);
+}
+```
+- TerminalsResponse: Response message 
+
+````protobuf
+message TerminalsResponse {
+  repeated Terminal terminals = 1;
+}
+message Terminal {
+  string terminal_id = 1; // terminal serial number
+  bool online = 2;
+}
+````
+3. **Create payment request**: Request to start the payment on the selected terminal.
+
+Key attributes to make a payment call:
 
-## Supported Python Versions
-- Python 3.6
-- Python 3.12
+- PayRequest : Request message to start payment.
+```protobuf
+message PayRequest {
+  string store_id = 1; // UUID of store
+  string amount = 2; // amount in BigDecimal/2.dp (0.00)
+  string terminal_id = 3; // send the payment to this terminal serial number
+  optional bool show_tips = 4; // Show tips on the terminal
+}
+```
+- KodyPayTerminalService : Service used to make the Payment request.
+- KodyPayTerminalService.Pay: RPC function used to make the Payment Request
+```protobuf
+service KodyPayTerminalService {
+  rpc Pay(PayRequest) returns (stream PayResponse);
+}
+```
+- PayResponse : Payment Response
+
+````protobuf
+message PayResponse {
+  PaymentStatus status = 1;
+  optional string failure_reason = 2; // only populated on failure
+  optional string receipt_json = 3; // json blob containing the receipt data
+  string order_id = 4;
+  google.protobuf.Timestamp date_created = 5;
+  optional string ext_payment_ref = 6;
+  google.protobuf.Timestamp date_paid = 7;
+  optional string total_amount = 8;
+  optional string sale_amount = 9;
+  optional string tips_amount = 10;
```

Comment:
```suggestion
  optional string tips_amount = 10; // tip amount paid, will be empty if no tips were selected.
```

Reply:
Can you verify whether it will be empty or `0.00` if no tips selected please 🙏🏻

Reply:
If we show these protobuf messages we should include comments to explain each field, or we convert them to python - and then include a bulleted list of explanations for each field.